WooCommerce/Classes/POS/Presentation/CartView.swift

```diff
@@ -195,9 +195,9 @@ private extension CartView {
 private extension CartView {
     var checkoutButton: some View {
         Button {
-            posModel.submitCart()
-            // Remove when totalsViewModel doesn't do the submission any more
-            cartViewModel.submitCart()
+            Task { @MainActor in
+                await posModel.submitCart()
+            }
```

Comment:
Do you think we should do the async task dispatch inside `submitCart/checkOut`? I'm not really sure but it seems messy to have this here.

Reply:
Good question. I had a similar back-and-forth when working with SwiftUI on a different feature. Mostly because the View needed to know about the results of an asynchronous action. However, if we want the ViewModel/AggregateModel to be the source of truth for the state, then it makes sense to dispatch it within the model. (Multiple) Views then could subscribe to relevant state changes when the cart is submitted. 

What are your thoughts on this? Do you have any other relevant caveats in mind?

Reply:
Yeah, I've not come across a situation where the view cares about the return of the action. That would be different, but I think we should probably be guided by the way a function's going to be used, and potentially offer both interfaces.

If there's something we want to do in a `task`, and cancel when the view's dismissed, exposing an async function from the aggregate model makes sense... but for a button tap, it might be better to hide that it's async and just let the view observe the state which is updated.

One counter-argument though... if we needed a button which showed a spinner when the action was underway, we could make an `AsyncButton` view. For that to work neatly, we would probably want the async function to be exposed. No-one wants the aggregate model to get littered with flags for assorted activity indicators.

I think a lot of those will be covered by the various state enums though: a `syncing` case can be observed from whichever parts of the UI need to consider showing spinners or being disabled.

Reply:
> I think a lot of those will be covered by the various state enums though: a syncing case can be observed from whichever parts of the UI need to consider showing spinners or being disabled.

Yes, exactly. If we implement what we need this way, then I don't see a need for async methods since these state variables could be observed.